src/plugins/embeddable/public/lib/embeddables/embeddable.tsx

```diff
@@ -253,21 +337,34 @@ export abstract class Embeddable<
 
     this.inputSubject.complete();
     this.outputSubject.complete();
+    this.destroyAPI();
 
     if (this.parentSubscription) {
       this.parentSubscription.unsubscribe();
     }
     return;
   }
 
+  public async untilInitializationFinished(): Promise<void> {
+    return new Promise((resolve) => {
```

Comment:
How about just `return lastValueFrom
 (this.initializationFinished)`

Reply:
How about using firstValueFrom then?

Reply:
can confirm that `lastValueFrom` worked great here! Thanks for the suggestion!

Reply:
I had to revert this change because of a jest test failure. I've opened a followup [issue](https://github.com/elastic/kibana/issues/175068) to see what went wrong here.